### PR TITLE
MGMT-4211 - Single node clusters will now use default master role for hosts

### PR DESCRIPTION
**Pull request by omertuc**
This improves validations because master validations are more strict than worker validations. See comments in PR diff for more information

### CODE REVIEW

**Review by osherdp — APPROVED** [review]
/lgtm

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -889,18 +889,23 @@ var _ = Describe("IgnitionParameters", func() {
...
+			ID:                   &clusterID,
+			Status:               swag.String(status),
+			HighAvailabilityMode: &highAvailabilityMode,
 		},
 	}
 	Expect(db.Create(c).Error).ToNot(HaveOccurred())
 	return c
 }
 
+func createCluster(db *gorm.DB, status string) *common.Cluster {
```
Comment (tsorya) [inline internal/bminventory/inventory_test.go:905]: not sure i understand why it is needed?

Comment (omertuc) [reply internal/bminventory/inventory_test.go:905]: Most tests don't care about availability mode, this is the function they call. It was there before, the diff makes it look like I added it.

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**